src/main/java/com/cloudant/search3/SearchServer.java

```diff
@@ -0,0 +1,273 @@
+import com.cloudant.search3.grpc.Search3.ErrorResponse;
+import com.cloudant.search3.grpc.Search3.ErrorResponse.Type;
+import com.cloudant.search3.grpc.Search3.GroupSearchRequest;
+import com.cloudant.search3.grpc.Search3.Index;
+import com.cloudant.search3.grpc.Search3.SearchRequest;
+import com.cloudant.search3.grpc.Search3.SetUpdateSeqRequest;
+import com.google.protobuf.Empty;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.MessageLite;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpContentCompressor;
+import io.netty.handler.codec.http.HttpContentDecompressor;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.codec.http.HttpServerExpectContinueHandler;
+import io.netty.handler.codec.http.HttpServerKeepAliveHandler;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.ssl.SslContext;
+import io.netty.util.concurrent.DefaultEventExecutorGroup;
+import io.netty.util.concurrent.EventExecutorGroup;
+
+public class SearchServer extends AbstractServer {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final String RPC_MESSAGE_TYPE = "rpc-message-type";
+
+    private static class SearchServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+
+        private static final ByteBuf EMPTY = wrappedBuffer(Empty.getDefaultInstance().toByteArray());
+
+        private final Search search;
+
+        public SearchServerHandler(final Search search) {
+            this.search = search;
+        }
+
+        @Override
+        public void channelReadComplete(final ChannelHandlerContext ctx) {
+            ctx.flush();
+        }
+
+        @Override
+        protected void channelRead0(final ChannelHandlerContext ctx, final HttpObject msg) throws Exception {
+            if (msg instanceof FullHttpRequest) {
+                final FullHttpRequest req = (FullHttpRequest) msg;
+
+                if (!POST.equals(req.method())) {
+                    ctx.writeAndFlush(error(Type.CLIENT_ERROR, req.uri() + " method not allowed"));
+                    return;
+                }
+
+                final String messageType;
+                final ByteBuf result;
+                switch (req.uri()) {
```

Comment:
indentation for all the `case` clauses?

Reply:
It looks right to me (and Eclipse fwiw), the nuance here is that I've put braces around the clause contents in order to reuse the variable names in each, for consistency.